### PR TITLE
Add title icon to `<DescriptionList`

### PR DESCRIPTION
**Pull request by dexterca**
AFTER
Screenshot 2023-08-09 at 3 14 21 PM

---

**NOTE**: This PR is merging into `feat-description-list` branch

### CODE REVIEW

File: src/components/DescriptionList/DescriptionListGroupHeading.tsx [hunk 1]
```diff
@@ -18,9 +18,13 @@ export const useStyles = makeStyles(
...
       gridColumn: '1/-1',
       letterSpacing: theme.pxToRem(1),
-      marginTop: theme.spacing(1),
       padding: theme.spacing(1.25, 0),
+      marginTop: theme.spacing(1),
       textTransform: 'uppercase',
+      '&:first-child, &:nth-child(2)': {
+        marginTop: 0,
+        paddingTop: 0,
+      },
```
Comment (dexterca) [inline src/components/DescriptionList/DescriptionListGroupHeading.tsx:27]: Remove top padding and top margin when section header appears at the top of a list or underneath a title.

File: src/components/DescriptionList/DescriptionList.stories.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ export default {
 } as ComponentMeta<typeof DescriptionList>;
 
 const Template: ComponentStory<typeof DescriptionList> = (args) => (
-  <DescriptionList title="Description List Title" {...args} />
```
Comment (dexterca) [inline src/components/DescriptionList/DescriptionList.stories.tsx:63]: Move title into it's own story.

File: src/components/DescriptionList/DescriptionList.tsx [hunk 3]
```diff
@@ -21,17 +21,31 @@ export const useStyles = makeStyles<DescriptionListProps>(
...
+        marginBottom: theme.spacing(0.5),
+      },
+      hasTitleIcon: {
+        alignItems: 'center',
+        display: 'grid',
+        gap: theme.spacing(1),
+        gridTemplateColumns: `${theme.pxToRem(22)} 1fr`,
+        '& ~ li': {
+          marginLeft: theme.pxToRem(30),
+        },
```
Comment (dexterca) [inline src/components/DescriptionList/DescriptionList.tsx:43]: I would like some help refining here so I don't have to use this `~` sibling selector. 

I would prefer to leverage `!!TitleIcon` within `DescriptionTerm.tsx` and `DescriptionListGroupHeading.tsx` but I couldn't figure out how to do that.

File: src/components/DescriptionList/DescriptionListGroupHeading.tsx [hunk 4]
```diff
@@ -18,9 +18,13 @@ export const useStyles = makeStyles(
       fontWeight: theme.typography.fontWeightBold,
       gridColumn: '1/-1',
       letterSpacing: theme.pxToRem(1),
-      marginTop: theme.spacing(1),
       padding: theme.spacing(1.25, 0),
+      marginTop: theme.spacing(1),
       textTransform: 'uppercase',
+      '&:first-child': {
+        marginTop: 0,
+        paddingTop: 0,
```
Comment (dexterca) [inline src/components/DescriptionList/DescriptionListGroupHeading.tsx:26]: Remove top padding and top margin when section header appears at the top of a list.

**Review by epeters3 — APPROVED** [review]

**Review by shawnzhu — APPROVED** [review]
👍

**Review by MorpheusNephew — APPROVED** [review]

**Pull request merged**